homeassistant/components/bluesound/button.py

```diff
@@ -0,0 +1,120 @@
+"""Button entities for Bluesound."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from pyblu import Player, SyncStatus
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.const import CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import (
+    CONNECTION_NETWORK_MAC,
+    DeviceInfo,
+    format_mac,
+)
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import BluesoundCoordinator
+from .media_player import DEFAULT_PORT
+from .utils import format_unique_id
+
+if TYPE_CHECKING:
+    from . import BluesoundConfigEntry
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: BluesoundConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Bluesound entry."""
+
+    async_add_entities(
+        [
+            SetSleepTimerButton(
+                config_entry.runtime_data.coordinator,
+                config_entry.runtime_data.player,
+                config_entry.data[CONF_PORT],
+            ),
+            ClearSleepTimerButton(
+                config_entry.runtime_data.coordinator,
+                config_entry.runtime_data.player,
+                config_entry.data[CONF_PORT],
+            ),
+        ],
+        update_before_add=True,
+    )
+
+
+def generate_device_info(sync_status: SyncStatus, port: int) -> DeviceInfo:
+    """Generate device info."""
+    if port == DEFAULT_PORT:
+        return DeviceInfo(
+            identifiers={(DOMAIN, format_mac(sync_status.mac))},
+            connections={(CONNECTION_NETWORK_MAC, format_mac(sync_status.mac))},
+            name=sync_status.name,
+            manufacturer=sync_status.brand,
+            model=sync_status.model_name,
+            model_id=sync_status.model,
+        )
+
+    return DeviceInfo(
+        identifiers={(DOMAIN, format_unique_id(sync_status.mac, port))},
+        name=sync_status.name,
+        manufacturer=sync_status.brand,
+        model=sync_status.model_name,
+        model_id=sync_status.model,
+        via_device=(DOMAIN, format_mac(sync_status.mac)),
+    )
+
+
+class SetSleepTimerButton(CoordinatorEntity[BluesoundCoordinator], ButtonEntity):
```

Comment:
Would it maybe make sense to create a base entity that will be inherited in both the media player and the buttons? It would help deduplicate the device info logic

Reply:
I would reommend doing this in a separate PR